Modules/overlapped.c

```diff
@@ -693,8 +753,32 @@ Overlapped_getresult(OverlappedObject *self, PyObject *args)
             if (transferred != PyBytes_GET_SIZE(self->allocated_buffer) &&
                 _PyBytes_Resize(&self->allocated_buffer, transferred))
                 return NULL;
+
             Py_INCREF(self->allocated_buffer);
             return self->allocated_buffer;
+        case TYPE_READ_FROM:
+            assert(PyBytes_CheckExact(self->read_from.buffer));
+
+            if (transferred != PyBytes_GET_SIZE(self->read_from.buffer) &&
+                _PyBytes_Resize(&self->read_from.buffer, transferred))
+            {
+                return NULL;
+            }
+
+            // unparse the address
+            addr = unparse_address((SOCKADDR*)&self->read_from.address,
+                                   self->read_from.address_length);
+
+            // The result is a two item tuple: (message, (address, port))
+            self->read_from.result = PyTuple_New(2);
+            // first item: message
+            PyTuple_SetItem(self->read_from.result, 0, self->read_from.buffer);
+            // second item: tuple(address, port)
+            PyTuple_SetItem(self->read_from.result, 1, addr);
```

Comment:
Please use `PyTuple_SET_ITEM()` instead of `PyTuple_SetItem()` when filling in a new tuple.